linera-storage/src/lib.rs

```diff
@@ -210,34 +208,22 @@ pub trait Storage: Sized {
             )
             .await?;
 
-        let epochs_and_committees = futures::future::try_join_all(
-            epoch_creation_events
-                .into_iter()
-                .filter_map(|index_and_event| {
-                    let epoch = Epoch::from(index_and_event.index);
-                    let maybe_blob_hash = bcs::from_bytes::<CryptoHash>(&index_and_event.event);
-                    epochs.contains(&epoch).then_some((epoch, maybe_blob_hash))
-                })
-                .map(|(epoch, maybe_blob_hash)| async move {
-                    let committee = read_committee(maybe_blob_hash?).await?;
-                    Result::<_, ViewError>::Ok((epoch, committee))
-                }),
-        )
-        .await?;
-
-        for (epoch, committee) in epochs_and_committees {
-            epochs.remove(&epoch);
-            result.insert(epoch, committee);
-        }
+        result.extend(
+            futures::future::try_join_all(
+                epoch_creation_events
+                    .into_iter()
+                    .take_while(|index_and_event| index_and_event.index <= max_epoch.0)
```

Comment:
I am always unsure whether the order of returning from the collection of futures is in the same order in which they were created 🤔 otherwise we might stop processing here prematurely (i.e. some future that resolves later on should still be returned).

Reply:
Ah, of course 🤦 I got confused by `read_events_from_index` which is an async function that returns a `Vec<_>`.